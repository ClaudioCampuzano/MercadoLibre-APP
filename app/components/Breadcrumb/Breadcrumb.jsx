import React, { Fragment } from 'react';
import { ChevronRight } from '@/public';
import styles from './Breadcrumb.module.scss';
import Image from 'next/image';

const Breadcrumb = ({ pathFromRoute = [] }) => {
  const current = pathFromRoute[pathFromRoute.length - 1];
  const previous = pathFromRoute.slice(0, pathFromRoute.length - 1);

  return (
    <nav className={styles.container}>
      {previous.map((label) => (
        <Fragment key={label}>
          <a className={styles.link} href={`/items?search=${label}`}>
            {label}
          </a>
          <Image
            src={ChevronRight}
            alt="Arrow"
            className={styles.imagenChevron}
            width="10"
            height="10"
          />
        </Fragment>
      ))}
      <span>{current}</span>
    </nav>
  );
};

export { Breadcrumb };
