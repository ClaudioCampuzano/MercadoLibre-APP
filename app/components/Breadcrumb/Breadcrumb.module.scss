@import '@/styles/constans.module.scss';

.container {
  color: $secondaryTextColor;
  margin-top: 10px;
  margin-bottom: 10px;
  width: 80%;
}

.imagenChevron {
  margin-left: 5px;
  margin-right: 5px;
}

.link {
  text-decoration: none;
  color: $secondaryTextColor;
}
