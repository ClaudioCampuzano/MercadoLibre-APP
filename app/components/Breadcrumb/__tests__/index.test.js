import React from 'react';

import { render } from '@testing-library/react';

import { Breadcrumb } from '../Breadcrumb';

const componentRender = (props = { pathFromRoute: ['first', 'second'] }) => {
  return render(<Breadcrumb {...props} />);
};

describe('Breadcrumb', () => {
  it('renders', () => {
    const { container } = componentRender();

    expect(container).toMatchSnapshot();
  });
});
