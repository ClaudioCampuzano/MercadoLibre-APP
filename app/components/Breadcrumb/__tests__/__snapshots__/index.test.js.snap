// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb renders 1`] = `
<div>
  <nav
    class="container"
  >
    <a
      class="link"
      href="/items?search=first"
    >
      first
    </a>
    <img
      alt="Arrow"
      class="imagenChevron"
      data-nimg="1"
      decoding="async"
      height="10"
      loading="lazy"
      src="/_next/image?url=%2Fimg.jpg&w=32&q=75"
      srcset="/_next/image?url=%2Fimg.jpg&w=16&q=75 1x, /_next/image?url=%2Fimg.jpg&w=32&q=75 2x"
      style="color: transparent;"
      width="10"
    />
    <span>
      second
    </span>
  </nav>
</div>
`;
